DatadogCore/Sources/Core/Storage/FilesOrchestrator.swift

```diff
@@ -58,7 +58,11 @@ internal class FilesOrchestrator: FilesOrchestratorType {
     }
 
     /// An extra information to include in metrics or `nil` if metrics should not be reported for this orchestrator.
-    let metricsData: MetricsData?
+    var metricsData: MetricsData?
+
+    /// Tracks number of pending batches in the track's directory
+    @ReadWriteLock
+    private var pendingBatches: Int = 0
```

Comment:
I'm using a lock here because getting and deleting a file can happen on different threads, but this can add penalty to the call thread.

The other option will be to read the directory each time we send the telemetry:
```swift
BatchDeletedMetric.pendingBatches: try? directory.files()
```
But this also has the drawback of reading the disk each time.